packages/next/src/next-devtools/dev-overlay/components/overview/segment-boundary-trigger.tsx

```diff
@@ -0,0 +1,339 @@
+import { useCallback, useState, useRef, useLayoutEffect } from 'react'
+import type { SegmentNodeState } from '../../../userspace/app/segment-explorer-node'
+
+export function SegmentBoundaryTrigger({
+  onSelectBoundary,
+  offset,
+}: {
+  onSelectBoundary: SegmentNodeState['setBoundaryType']
+  offset: number
+}) {
+  const [isOpen, setIsOpen] = useState(false)
+  const [dropdownPosition, setDropdownPosition] = useState({ top: 0, right: 0 })
+  const triggerRef = useRef<HTMLButtonElement>(null)
+
+  const updateDropdownPosition = useCallback(() => {
+    if (triggerRef.current) {
+      const rect = triggerRef.current.getBoundingClientRect()
+      setDropdownPosition({
+        top: rect.bottom + offset,
+        right: window.innerWidth - rect.right,
+      })
+    }
+  }, [offset])
+
+  useLayoutEffect(() => {
+    if (isOpen) {
+      updateDropdownPosition()
+    }
+  }, [isOpen, updateDropdownPosition])
+
+  const handlePointerEnter = useCallback(() => {
+    setIsOpen(true)
+    updateDropdownPosition()
+  }, [updateDropdownPosition])
+
+  const handlePointerLeave = useCallback(
+    (e: React.PointerEvent<HTMLElement>) => {
+      const relatedTarget = e.relatedTarget
+      if (
+        relatedTarget instanceof HTMLElement &&
+        (relatedTarget.closest(
+          '[data-nextjs-dev-overlay-segment-boundary-dropdown-backdrop]'
+        ) ||
+          relatedTarget.closest(
+            '[data-nextjs-dev-overlay-segment-boundary-trigger-button]'
+          ))
+      ) {
+        return
+      }
+
+      // This will only auto close for mouse leave,
+      // but preserve for touch events.
+      if (e.pointerType === 'mouse') {
+        setIsOpen(false)
+      }
+    },
+    []
+  )
+
+  const triggerOptions = [
+    { label: 'Trigger Loading', value: 'loading', icon: <LoadingIcon /> },
+    { label: 'Trigger Error', value: 'error', icon: <ErrorIcon /> },
+    { label: 'Trigger Not Found', value: 'not-found', icon: <NotFoundIcon /> },
+  ]
+
+  const resetOption = {
+    label: 'Reset',
+    value: 'reset',
+    icon: <ResetIcon />,
+  }
+
+  const handleSelect = (value: string) => {
+    if (value === 'not-found') {
+      onSelectBoundary('not-found')
+    } else if (value === 'loading') {
+      onSelectBoundary('loading')
+    } else if (value === 'error') {
+      onSelectBoundary('error')
+    } else if (value === 'reset') {
+      onSelectBoundary(null)
+    }
+    setIsOpen(false)
+  }
+
+  return (
+    <div className="segment-boundary-trigger">
+      {/* add the reset button here as well, use the same icon */}
+      <button
+        className="segment-boundary-trigger-button segment-boundary-trigger-button--reset"
+        onClick={() => handleSelect(resetOption.value)}
```

Comment:
The reset affordance only working for "Trigger loading" is expected?

Reply:
It's the issue we discussed about reseting error boundary, will be covered in the future PRs